session.go

```diff
@@ -218,7 +218,12 @@ const (
 //     mongodb://myuser:mypass@localhost:40001,otherhost:40001/mydb
 //
 // If the port number is not provided for a server, it defaults to 27017.
+// A host could be an Unix Domain Socket (POSIX Local IPC Socket) like:
 //
+//     /var/run/mongodb/mongod.sock
```

Comment:
I had a quick look, and the [mongodb connection string specification](https://github.com/mongodb/specifications/blob/master/source/connection-string/connection-string-spec.rst#host) states:

``` 
UNIX domain sockets MUST end in ".sock" and MUST be URL encoded, for
example:

	mongodb://user:pass@%2Ftmp%2Fmongodb-27017.sock/authDB?replicaSet=rs

The host information cannot contain an unescaped slash ("/"), if it does then an
exception MUST be thrown informing users that paths must be URL encoded. For
example:

	Unsupported host '/tmp/mongodb-27017.sock', UNIX socket domain paths must be
	URL encoded.
```

Which seems weird, but I'm sure there's a good reason.